src/client/clientenvironment.cpp

```diff
@@ -170,9 +170,21 @@ void ClientEnvironment::step(float dtime)
 						lplayer->physics_override_gravity * dtime_part * 2.0f;
 
 				// Liquid floating / sinking
-				if (lplayer->in_liquid && !lplayer->swimming_vertical &&
-						!lplayer->swimming_pitch)
-					speed.Y -= lplayer->movement_liquid_sink * dtime_part * 2.0f;
+				if (lplayer->in_liquid && !lplayer->swimming_pitch) {
```

Comment:
Here's something that bothers me, sinking only happens when pitch move mode is off, that seems wrong, maybe the second condition should be removed?